test/jasmine/tests/axes_test.js

```diff
@@ -5196,6 +5196,301 @@ describe('Test axes', function() {
             });
         });
     });
+
+    describe('label positioning using *ticklabelmode*: "period"', function() {
+        var gd;
+
+        beforeEach(function() {
+            gd = createGraphDiv();
+        });
+
+        afterEach(destroyGraphDiv);
+
+        function _assertPositions(msg, exp) {
+            var ax = gd._fullLayout.xaxis;
+            var positions = ax._vals.map(function(d) { return ax.c2d(d.periodX); });
+            expect(positions).withContext(msg).toEqual(exp);
+        }
+
+        function _assertLabels(msg, exp) {
+            var ax = gd._fullLayout.xaxis;
+            var labels = ax._vals.map(function(d) { return d.text; });
+            expect(labels).withContext(msg).toEqual(exp);
+        }
+
+        ['%Y', '%y'].forEach(function(tickformat, i) {
+            it('should respect yearly tickformat that includes ' + tickformat, function(done) {
+                Plotly.newPlot(gd, {
+                    data: [{
+                        x: ['2020-01-01', '2026-01-01']
+                    }],
+                    layout: {
+                        width: 1000,
+                        xaxis: {
+                            ticklabelmode: 'period',
+                            tickformat: tickformat
+                        }
+                    }
+                })
+                .then(function() {
+                    _assertPositions('', [
+                        '2019-07-02 15:00',
+                        '2020-07-01 15:00',
+                        '2021-07-02 15:00',
+                        '2022-07-02 15:00',
+                        '2023-07-02 15:00',
+                        '2024-07-01 15:00',
+                        '2025-07-02 15:00',
+                        '2026-07-02 15:00'
+                    ]);
+                })
+                .then(function() {
```

Comment:
```suggestion

```

Not a big deal, but here (and the other new tests) these assertions can be done together.

Reply:
Done in https://github.com/plotly/plotly.js/pull/5065/commits/1c94ba99bbe3c9ad7bfd05418f6082b0946a88fb.